web/modules/custom/joinup_user/joinup_user.module

```diff
@@ -304,6 +311,34 @@ function joinup_user_form_alter(&$form, FormStateInterface $form_state, $form_id
   $form['#validate'][] = 'joinup_user_form_alter_validate';
 }
 
+/**
+ * Implements joinup_user_FORM_ID_alter().
+ */
+function joinup_user_form_user_register_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['actions']['submit']['#submit'][] = 'joinup_user_submit_spam_guard';
+}
+
+/**
+ * Represents an additional submit callback for the user registration form.
+ *
+ * @param array $form
+ *   The form render array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ */
+function joinup_user_submit_spam_guard(array $form, FormStateInterface $form_state) {
+  $user_id = $form_state->getFormObject()->getEntity()->id();
```

Comment:
I couldn't test practically, but can't we use the `$form_state->get('user')` (see `web/core/modules/user/src/RegisterForm.php:96`) ?

Reply:
Well, initially I did `$user = $form_state->getFormObject()->getEntity()`, then on `$user->delete()` but I received the exception:

```
Error: Call to a member function get() on null in pathauto_entity_delete() (line 108 of modules/contrib/pathauto/pathauto.module).
```
Same happens with `$form_state->get('user')`. The reason was that `$entity->get('path')` is `NULL` in this stage because it seems that `path` fiels is not yet populated as the entity building in not over yet. But an entity fill load gives a full formed entity.